src/caja-desktop-window.c

```diff
@@ -230,15 +303,21 @@ unrealize (GtkWidget *widget)
     CajaDesktopWindow *window;
     CajaDesktopWindowPrivate *details;
     GdkWindow *root_window;
+    GdkDisplay *display;
 
     window = CAJA_DESKTOP_WINDOW (widget);
     details = window->details;
+    display = gdk_screen_get_display (gdk_screen_get_default());
 
-    root_window = gdk_screen_get_root_window (
+    /*Avoid root window on wayland-it's not supposed to work*/
+    if  (GDK_IS_X11_DISPLAY (display))
+    {
+        root_window = gdk_screen_get_root_window (
                       gtk_window_get_screen (GTK_WINDOW (window)));
 
-    gdk_property_delete (root_window,
+        gdk_property_delete (root_window,
```

Comment:
Indentation

Reply:
That one moves into an   `  if (GDK_IS_X11_DISPLAY (display))` clause, thus the indent change to match the rest of that clause

Reply:
Sorry, my fault. But the next Line should be moved too or not.... :)

Reply:
First letter of 2nd line under first letter of 1st line is how that is aligned now, and how similar things elsewhere in the code are formatted

Reply:
That looks good to me now